src/Dofs/ConstraintHandler.jl

```diff
@@ -315,14 +315,14 @@ function _local_face_dofs_for_bc(interpolation, field_dim, components, offset, b
     return local_face_dofs, local_face_dofs_offset
 end
 
-# Dirichlet on nodeset
-function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcnodes::Set{Int}, interpolation::Interpolation, field_dim::Int, offset::Int, bcvalue::BCValues, cellset::Set{Int}=Set{Int}(1:getncells(ch.dh.grid)))
-    if interpolation !== default_interpolation(typeof(ch.dh.grid.cells[first(cellset)]))
+function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcnodes::Set{Int}, interpolation::Interpolation, field_dim::Int, offset::Int, bcvalue::BCValues, cellset::Set{Int}=Set{Int}(1:getncells(getgrid(ch.dh))))
+    grid = getgrid(ch.dh)
+    if interpolation !== default_interpolation(getcelltype(grid, first(cellset)))
```

Comment:
`getcelltype(grid, i)` should probably be typeof(getcell(grid, i))`. Although, `getcelltype` can be optimized for homogeneous grids, but not sure that matters.

Reply:
I added it because I thought it is a nice convenience which is clearer in the intent.